addons/mail/static/tests/helpers/mock_models.js

```diff
@@ -95,6 +95,15 @@ class MockModels {
                 },
                 records: [],
             },
+            'mail.channel.partner': {
+                fields: {
+                    channel_id: { string: "Channel", type: 'many2one', relation: 'mail.channel' },
+                    fetched_message_id: { string: "Last Fetched", type: 'many2one', relation: 'mail.message' },
+                    partner_id: { string: "Partner", type: 'many2one', relation: 'res.partner', default() { return this.currentPartnerId; } },
+                    seen_message_id: { string: "Last Fetched", type: 'many2one', relation: 'mail.message' },
+                },
+                records: [],
+            },
```

Comment:
If we start to introduce `channel.partner` in mock server (and I understand the need), then we'll need to move a lot of stuff into it (everything that was in channel for simplicity) to keep the mock server consistent.

That includes at least: `is_pinned,` `fetched_message_id`, `is_minimized`, `state`, `is_moderator`, `message_unread_counter`, `seen_message_id`.

This is a big change that will require diff on all the tests pretty much... So we can create a new task for it, but it should be done ASAP.

In this PR, you at least need to remove from channel those that you redefine here.

Reply:
We could also edit mock server to write in channels instead ? Even if it's not really good ..

Reply:
The need for channel_partner is to simulate its data for different users in parallel.

If you introduce it but only use it for the current partner then it's not really needed, and we should keep updating the channel value for simplicity.

Reply:
I'm pretty sure it's not only for current user